ament_pclint/ament_pclint/main.py

```diff
@@ -0,0 +1,372 @@
+
+    parser = argparse.ArgumentParser(
+        description='Perform static code analysis using pclint.',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument(
+        'paths',
+        nargs='*',
+        default=[os.curdir],
+        help='Files and/or directories to be checked. Directories are searched recursively for '
+             'files ending in one of %s.' %
+             ', '.join(["'.%s'" % e for e in extensions]))
+    parser.add_argument(
+        '--language',
+        help="Force pclint to analyze files as specified language. Supported languages c or cpp")
+    if os.name == 'nt':
+        parser.add_argument(
+            '--pclint-config-dir',
+            nargs=1,
+            type=str,
+            default=[os.path.dirname(os.path.realpath(__file__)) + "/config/msvc"],
+            help='Specify a different configuration directory. Default path is config/msvc')
+    else:
+        parser.add_argument(
+            '--pclint-config-dir',
+            nargs=1,
+            type=str,
+            default=[os.path.dirname(os.path.realpath(__file__)) + "/config/gcc"],
+            help='Specify a different configuration directory. Default path is config/gcc')
+    parser.add_argument(
+        '--pclint-config-file',
+        nargs=1,
+        type=str,
+        help='Specify a different configuration file. You may need to use --pclint-config-dir.')
+    parser.add_argument(
+        '--include-directories',
+        nargs='*',
+        help='Paths to directories that are included in the files that should be analyzed')
+    parser.add_argument(
+        '--compiler-definitions',
+        nargs='*',
+        default='',
+        help='Compiler definitions that are necessary for PCLint.')
+    # not using a file handle directly
+    # in order to prevent leaving an empty file when something fails early
+    parser.add_argument(
+        '--xunit-file',
+        help='Generate a xunit compliant XML file')
+    parser.add_argument(
+        '--debug',
+        action='store_true',
+        help='Print the full pclint command that is being called'
+    )
+    args = parser.parse_args(argv)
+
+    if args.xunit_file:
+        start_time = time.time()
+
+    files = get_files(args.paths, extensions)
+    if not files:
+        print('No files found', file=sys.stderr)
+        return 1
+
+    additional_paths = None
+    pclint_bin = find_executable('pclp64', additional_paths=additional_paths)
+    if not pclint_bin:
+        print("Could not find pclint executable 'pclp64'", file=sys.stderr)
+        return 1
+
+    # Prepare pclint command
+    base_cmd = [pclint_bin,
+                '-i"%s"' % args.pclint_config_dir[0],
+                '-i"%s"' % os.path.join(os.path.dirname(os.path.realpath(__file__)), "config"),
+                'env-xml.lnt']
+
+    # Add include directories from arguments
+    if args.include_directories is not None:
+        for directory in args.include_directories:
```

Comment:
Suggestion to avoid the extra conditional:

```
for directory in (args.include_directories or []):
```

Same below.

Reply:
Done